connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundConnectorContextImpl.java

```diff
@@ -94,14 +99,28 @@ public InboundConnectorDefinition getDefinition() {
 
   @Override
   public void reportHealth(Health health) {
-    this.health = health;
+    if (this.health == null) {
+      this.health = health;
+      return;
+    }
+    this.health = this.health.merge(health);
```

Comment:
Can we use and if/else statement here instead of an if combined with an early return?

Reply:
@sbuettner Sure, but I think the early return can reduce complexity. In general in this kind of case, if in the future we'll add a new condition inside the else branch that would result in nested if-else statements which I would avoid. If we implement this now with early return, we make it easier to eliminate nested if-else statements in the future. I'll gladly rewrite this, but please elaborate on why you think that's better.